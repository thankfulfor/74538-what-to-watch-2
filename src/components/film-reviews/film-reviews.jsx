import * as React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {formatDate} from '../../utils/date-convert.js';

export const FilmReviews = (props) => {

  const {reviews} = props;

  if (reviews === undefined) {
    return null;
  }

  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">
        {reviews.length === 0
          ? <p className="review__text">There is no reviews. Be the first to add one.</p>
          : reviews.map((review) =>
            <div className="review" key={review.user.name}>
              <blockquote className="review__quote">
                <p className="review__text">{review.comment}</p>

                <footer className="review__details">
                  <cite className="review__author">{review.user.name}</cite>
                  <time className="review__date" dateTime="2016-12-24">{formatDate(review.date)}</time>
                </footer>
              </blockquote>

              <div className="review__rating">{review.rating}</div>
            </div>
          )
        }
      </div>
    </div>
  );
};

FilmReviews.propTypes = {
  reviews: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    reviews: state.reviews,
  });
};

export default connect(mapStateToProps)(FilmReviews);
