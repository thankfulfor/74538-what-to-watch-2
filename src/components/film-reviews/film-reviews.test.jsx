import React from 'react';
import renderer from 'react-test-renderer';
import {FilmReviews} from './film-reviews.jsx';
import {BrowserRouter} from 'react-router-dom';

const reviews = [{
  comment: `This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.`,
  date: `2019-11-28T08:42:04.899Z`,
  id: 1,
  rating: 2.1,
  user: {id: 11, name: `Jack`}
}];

it(`FilmReviews корректно рендерится после перезапуска`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FilmReviews reviews={reviews} />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
