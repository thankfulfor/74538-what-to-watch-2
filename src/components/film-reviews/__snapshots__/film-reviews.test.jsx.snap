// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmReviews корректно рендерится после перезапуска 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            November 28, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2.1
      </div>
    </div>
  </div>
</div>
`;
